process/smartContract/hooks/blockChainHook.go

```diff
@@ -480,6 +494,42 @@ func (bh *BlockChainHookImpl) SetCurrentHeader(hdr data.HeaderHandler) {
 	bh.mutCurrentHdr.Unlock()
 }
 
+// SaveCompiledCode saves to cache and storage the compiled code
+func (bh *BlockChainHookImpl) SaveCompiledCode(codeHash []byte, code []byte) {
+	bh.compiledScPool.Put(codeHash, code, len(code))
+	err := bh.compiledScStorage.Put(codeHash, code)
+	if err != nil {
+		log.Debug("SaveCompiledCode", "error", err)
```

Comment:
you can also log the `codeHash` for debug reasons

Reply:
this will happen only if we have a problem with storage - codehash does not help. but added.

Reply:
yes, but at least we can know that for a given hash the save failed